p2p/swarm.go

```diff
@@ -370,9 +372,12 @@ func (s *swarm) RegisterDirectProtocol(protocol string) chan service.DirectMessa
 }
 
 // RegisterGossipProtocol registers an handler for gossip based `protocol`
-func (s *swarm) RegisterGossipProtocol(protocol string) chan service.GossipMessage {
+func (s *swarm) RegisterGossipProtocol(protocol string, prio int) chan service.GossipMessage {
 	mchan := make(chan service.GossipMessage, s.config.BufferSize)
 	s.protocolHandlerMutex.Lock()
+	if err := s.pq.Set(protocol, prio, cap(mchan)); err != nil {
```

Comment:
why does prio-q needs to be aware of protocols ? just let it have per-determined set of prios

Reply:
See prev comment.